### PR TITLE
Add integration tests for token attribute filtering, verified_claims validation, and prompt=none handling

### PR DESCRIPTION
**Pull request by AmshikaH**
### Purpose

Adds integration tests for four previously under-covered areas for OAuth protocol & token issuance:

- Client_credentials and agent token claim content (group/role/entity/system claims, allow-list scoping)
- Nested claims support: OIDC Identity Assurance `verified_claims` request parameter validation
- Attribute allow-list filtering across the `ciba`, `refresh_token`, and `token_exchange` grant paths
- The `prompt=none` error path at `/authorize` and `/oauth2/par`

### Approach

 - `tests/integration/oauth/token/cc_client_attributes_test.go` and `tests/integration/agent/agent_client_attributes_test.go` verify that OU, groups, roles, schema, and system claims are correctly included/excluded from issued tokens based on the configured attribute allow-list, for both client_credentials apps and agents. Added a `Token` field to `OAuthAgentConfig` in `tests/integration/agent/model.go` (mirroring the existing pattern in `tests/integration/application/model.go`) so agent tests can configure an access-token attribute allow-list.
- `tests/integration/oauth/claims/verified_claims_test.go` covers the `claims` request parameter's `verified_claims` (OIDC IDA) object and array forms, including rejection of malformed `verification`/`trust_framework` shapes and conflicting `value`/`values` constraints. Note: `verified_claims` requests are validated but not currently resolved into output token/userinfo claims, so these tests assert request acceptance/rejection rather than claim presence.
- `tests/integration/oauth/ciba/ciba_attribute_filter_test.go`, `tests/integration/oauth/token/refresh_token_attribute_filter_test.go`, and `tests/integration/oauth/token/token_exchange_attribute_filter_test.go` each verify that the subject-attribute allow-list is enforced consistently for the CIBA, refresh_token, and token_exchange grant types, not just authorization_code/client_credentials.
- `tests/integration/oauth/authz/prompt_none_test.go` and a new case in `tests/integration/oauth/par/par_test.go` cover `prompt=none` handling: `login_required` when no session exists, `invalid_request` when combined with other prompt values, and rejection of unsupported prompt values. The PAR case documents current behavior as-is (see comment in the test).
- Added a `prompt` parameter to the shared `initiateAuthorizationFlow` helper (`tests/integration/testutils/oauth2_utils.go`) plus a public `InitiateAuthorizationFlowWithPrompt` wrapper, so tests can drive the authorize endpoint with an explicit `prompt` value without changing any existing callers' behavior (all pass `""`).

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] Followed the contribution guidelines.
- [ ] Manual test round performed and verified.
- [ ] Documentation provided. (Add links if there are any)
    - [ ] Ran Vale and fixed all errors and warnings
- [ ] Tests provided. (Add links if there are any)
    - [ ] Unit Tests
    - [ ] Integration Tests
- [ ] Breaking changes. (Fill if applicable)
    - [ ] Breaking changes section filled.
    - [ ] `breaking change` label added.

### Security checks
- [ ] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

## Summary by CodeRabbit

* **Tests**
  * Expanded OAuth coverage for access-token attribute filtering across client credentials, refresh tokens, token exchange, CIBA, and agent flows.
  * Added validation confirming allowed claims are retained while non-allowed claims are excluded.
  * Added OIDC Identity Assurance verified-claims coverage for valid object and array formats, plus invalid-request scenarios.
  * Added authorization coverage for `prompt=none`, unsupported values, invalid combinations, redirects, and state handling.
  * Improved authorization-flow test support for the `prompt` parameter.

### CODE REVIEW

File: tests/integration/oauth/claims/verified_claims_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+// TestVerifiedClaims_EmptyClaims_InvalidRequest verifies that a verified_claims entry with an
+// empty claims object (requesting nothing) is rejected as invalid_request.
+func (ts *ClaimsParameterTestSuite) TestVerifiedClaims_EmptyClaims_InvalidRequest() {
+	claimsParam := `{"userinfo":{"verified_claims":{
+		"verification":{"trust_framework":"de_aml"},
+		"claims":{}
+	}}}`
+
+	ts.assertVerifiedClaimsRejected(claimsParam)
+}
```
[the remark this answers is not in the document]
Comment (AmshikaH) [reply tests/integration/oauth/claims/verified_claims_test.go:69]: Fixed with [1a19fc7](https://github.com/thunder-id/thunderid/pull/4887/commits/1a19fc7af85bed4b3a13fbed28bbdea110ea91fd).

File: tests/integration/oauth/ciba/ciba_attribute_filter_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+	callbackStatus, _ := ts.cibaPostCallback(bcResp.AuthReqID, flowStep.Assertion)
+	ts.Require().Equal(http.StatusOK, callbackStatus)
+
+	tokenRes := ts.cibaPollTokenAs(bcResp.AuthReqID, "", cibaAttrFilterClientID, cibaAttrFilterClientSecret)
+	if tokenRes.statusCode == http.StatusBadRequest && tokenRes.errorCode == "slow_down" {
+		time.Sleep(cibaPollIntervalSeconds * time.Second)
+		tokenRes = ts.cibaPollTokenAs(bcResp.AuthReqID, "", cibaAttrFilterClientID, cibaAttrFilterClientSecret)
+	}
+	ts.Require().Equal(http.StatusOK, tokenRes.statusCode, "AUTHENTICATED request should issue tokens")
```
[the remark this answers is not in the document]
Comment (AmshikaH) [reply tests/integration/oauth/ciba/ciba_attribute_filter_test.go]: Fixed.

File: tests/integration/oauth/ciba/ciba_attribute_filter_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	ts.Require().Equal(http.StatusOK, callbackStatus)
+
+	var tokenRes cibaTokenResult
+	for attempt := 0; attempt < 3; attempt++ {
+		tokenRes = ts.cibaPollTokenAs(bcResp.AuthReqID, "", cibaAttrFilterClientID, cibaAttrFilterClientSecret)
+		if tokenRes.statusCode != http.StatusBadRequest || tokenRes.errorCode != "slow_down" {
+			break
+		}
+		time.Sleep(cibaPollIntervalSeconds * time.Second)
+	}
```
[the remark this answers is not in the document]
Comment (AmshikaH) [reply tests/integration/oauth/ciba/ciba_attribute_filter_test.go]: Fixed with _[67d7c5f](https://github.com/thunder-id/thunderid/pull/4887/commits/67d7c5f10ee41d82f1e02654bba9ee7f16f952e9)._

**Comment by AmshikaH** [comment]
@coderabbitai review

**Comment by AmshikaH** [comment]
@coderabbitai resume

**Review by sahandilshan — APPROVED** [review]
